packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <label ref="placeholder" class="placeholder"></label>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { searchBoxXModule } from '../x-module';
+  import SearchInput from './search-input.vue';
+  @Component({
+    components: { SearchInput },
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    @State('searchBox', 'query')
+    public query!: string;
+
+    @Prop({ default: () => [''] })
+    protected placeholdersArray!: string[];
+
+    protected get isQueryEmpty(): boolean {
+      return this.query.length === 0;
+    }
+
+    @XOn(['UserHoveringOverSearchBox'])
+    animatePlaceholder(): void {
+      if (this.isQueryEmpty) {
+        var array = this.placeholdersArray;
+        var element = this.$refs.placeholder as HTMLElement;
+        element.classList.add('visible');
+        for (let i = 0; i < array.length; i++) {
+          setTimeout(function () {
+            element.innerText = array[i];
+            if (i === array.length - 1) {
+              element.classList.remove('visible');
+            }
+          }, i * 2000);
+        }
+      }
+    }
```

Comment:
Currently the animation skips the last value and is not running on a loop 😅. Instead of setTimeout you should use setInterval and clear the interval when the input is not hovered. 

In order to loop through the array elements you can store the active element index, check if the next one exists and if not then set the active index to 0. This way the elements will be changed in an infinite loop. 

Take this just as an idea:

```
let activeIndex = 0;

let activePlaceholder = 0;

setInterval(() => {
const next = activeIndex++;
activePlaceholder = placeholders[next] ? next : 0;
})
```